src/core/BufferPool.cpp

```diff
@@ -24,41 +24,39 @@
  *
  */
 
-#include "BufferManager.h"
+#include "BufferPool.h"
 
-#include "Engine.h"
-#include "Mixer.h"
-#include "MemoryManager.h"
+#include <cstring>
+#include "MemoryPool.h"
 
-static fpp_t framesPerPeriod;
+static std::unique_ptr<_MemoryPool_Base> pool;
+const int BM_INITIAL_BUFFERS = 256;
 
-void BufferManager::init( fpp_t framesPerPeriod )
+void BufferPool::init( fpp_t framesPerPeriod )
 {
-	::framesPerPeriod = framesPerPeriod;
+	pool.reset(new _MemoryPool_Base(framesPerPeriod * sizeof(sampleFrame), BM_INITIAL_BUFFERS));
 }
 
-
-sampleFrame * BufferManager::acquire()
+sampleFrame * BufferPool::acquire()
 {
-	return MM_ALLOC( sampleFrame, ::framesPerPeriod );
+	return reinterpret_cast<sampleFrame*>(pool->allocate());
 }
 
-void BufferManager::clear( sampleFrame *ab, const f_cnt_t frames, const f_cnt_t offset )
+void BufferPool::clear( sampleFrame *ab, const f_cnt_t frames, const f_cnt_t offset )
```

Comment:
Could we pull this function out of `BufferPool`? It doesn't look like a functionality of pool.

Reply:
I think we should, but I'm not sure where to.

Reply:
I'm not sure but how about `MixHelpers`?

Reply:
For references, #2383 moved the function from `Mixer` to `BufferManager`.